packages/design-system/src/components/keyboard/keyboard.tsx

```diff
@@ -0,0 +1,233 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import {
+  useEffect,
+  useState,
+  useContext,
+  useBatchingCallback,
+  useCallback,
+} from '@googleforcreators/react';
+import type { DependencyList } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import type {
+  KeyEffectCallback,
+  KeyNameOrSpec,
+  RefOrNode,
+} from '../../types/keyboard';
+import Context from './context';
+import {
+  getNodeFromRefOrNode,
+  getOrCreateMousetrap,
+  HTMLElementWithMouseTrap,
+  resolveKeySpec,
+  createKeyHandler,
+  createShortcutAriaLabel,
+  prettifyShortcut,
+} from './utils';
+
+const globalRef: { current: null | HTMLElement } = { current: null };
+
+function setGlobalRef() {
+  if (!globalRef.current) {
+    globalRef.current = document.documentElement;
+  }
+}
+
+/**
+ * See https://craig.is/killing/mice#keys for the supported key codes.
+ */
+function useKeyEffectInternal(
+  refOrNode: RefOrNode,
+  keyNameOrSpec: KeyNameOrSpec,
+  type: string | undefined,
+  callback: KeyEffectCallback,
+  deps: DependencyList
+) {
+  const { keys } = useContext(Context);
+  //eslint-disable-next-line react-hooks/exhaustive-deps -- Pass through provided deps.
+  const batchingCallback = useBatchingCallback(callback, deps || []);
+  useEffect(
+    () => {
+      const nodeEl = getNodeFromRefOrNode(refOrNode);
+      if (!nodeEl) {
+        return undefined;
+      }
+      if (
+        nodeEl.nodeType !== /* ELEMENT */ 1 &&
+        nodeEl.nodeType !== /* DOCUMENT */ 9
```

Comment:
Assuming this works:

```suggestion
        nodeEl.nodeType !== Node.ELEMENT_NODE &&
        nodeEl.nodeType !== Node.DOCUMENT_NODE
```